front/components/vaults/VaultUpsertFolderModal.tsx

```diff
@@ -18,24 +18,28 @@ import { useContext, useState } from "react";
 import { SendNotificationsContext } from "@app/components/sparkle/Notification";
 import type { PostVaultDataSourceResponseBody } from "@app/pages/api/w/[wId]/vaults/[vId]/data_sources";
 
-export default function VaultCreateFolderModal({
+export default function VaultUpsertFolderModal({
   isOpen,
   setOpen,
   owner,
   vault,
   dataSources,
+  folder,
 }: {
   isOpen: boolean;
   setOpen: (isOpen: boolean) => void;
   owner: WorkspaceType;
   vault: VaultType;
   dataSources: DataSourceType[];
+  folder: DataSourceType | null;
 }) {
   const router = useRouter();
   const sendNotification = useContext(SendNotificationsContext);
 
-  const [name, setName] = useState<string | null>(null);
-  const [description, setDescription] = useState<string | null>(null);
+  const [name, setName] = useState<string | null>(folder?.name ?? null);
+  const [description, setDescription] = useState<string | null>(
+    folder?.description ?? null
+  );
```

Comment:
If you change description, cancel and re-edit, is it ok to keep the "edited" description ?

Reply:
If we cancelled it should be reset to the db value

Reply:
Fixed, thanks!